File: services/media/internal/domain/music.go
package domain

import (
	"time"
	"uuid"
)

// MusicTrack описывает музыкальный трек, связанный с audio media-объектом.
type MusicTrack struct {
	ID        uuid.UUID
	MediaID   uuid.UUID
	UserID    uuid.UUID
	Title     string
	Artist    string
	Duration  *float64
	CreatedAt time.Time
}

// CanBeManagedBy проверяет, принадлежит ли трек указанному пользователю.
func (t MusicTrack) CanBeManagedBy(userID uuid.UUID) bool {
	return t.UserID == userID
}

File: frontend/src/entities/music/model/music.ts
export type MusicTrack = {
  id: string;
  media_id: string;
  user_id: string;
  title: string;
  artist: string;
  duration: number | null;
  created_at: string;
  is_saved: boolean;
};

export type MusicResponse = {
  tracks: MusicTrack[];
};
